bin/cylc-get-host-metrics

```diff
@@ -154,19 +137,41 @@ def process_load():
     String keys e.g. 'load:1' give minutes average was measured over.
     """
     cmd, av_times, patt = LOAD_TEMPLATES
-    av_raw_tuple = extract_pattern(patt, run_command_string(cmd))
+    av_raw_tuple = extract_pattern(patt, run_command(cmd))
     av_floats_tuple = [float(av_raw) for av_raw in av_raw_tuple]
     return dict(zip(
         ["load:" + str(time) for time in av_times], av_floats_tuple))
 
 
 def process_memory():
-    """Return dict with integer usable memory value from '/proc/meminfo'."""
-    cmd, patt = MEMORY_TEMPLATES
-    mem_tuple = extract_pattern(patt, run_command_string(cmd))
-    # Sum free, buffer and cache memory values (free plus/minus buffer/cache).
-    usable_memory = sum([int(mem_val) for mem_val in mem_tuple])
-    return {"memory": usable_memory}
+    """Return available memory.
+
+    Note the `free` command is inconsistent accross Linux distros.
+
+    Extracts the `MemAvailable` field from `/proc/meminfo` if present else
+    returns the sum of `MemFree`, `Buffers` and `Cached`.
+
+    ('free -k' command inconsistent across Linux distros, see e.g.
+     'https://access.redhat.com/solutions/406773', so check '/proc' directly)
+
+    """
+    stdout = run_command(['cat', '/proc/meminfo'])
+    fields = dict((
+        line.split()[:2]
+        for line in stdout.splitlines()
+    ))
+    try:
+        mem = int(fields['MemAvailable:'])
+    except KeyError:
+        # fallback for older kernels - provide our own estimate
+        # see: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/
+        #      linux.git/commit/?id=34e431b0ae398fc54ea69ff85ec700722c9da773
+        mem = sum(
+            int(fields['MemFree:']),
+            int(fields['Buffers:']),
+            int(fields['Cached:'])
+        )
+    return {'memory': mem}
```

Comment:
This function could be tested with mock (I think would just need to mock `run_command`).

The except block seems to have a bug as `sum` is being called as `sum(int, int, int)` I think? Running quick checks with IDE and it alerted for this one.

Reply:
(oh, I think it could be `sum([int, int, int, ...])` instead)

Reply:
Good spot!